recipes-bsp/tegra-binaries/tegra186-flashtools-native_32.1.0.bb

```diff
@@ -40,13 +54,26 @@ do_install() {
     install -m 0755 ${S}/bootloader/nv_smd_generator ${D}${BINDIR}
     install -m 0755 ${S}/bootloader/BUP_generator.py ${D}${BINDIR}
     sed -i -e's,^#!/usr/bin/python,#!/usr/bin/env python,' ${D}${BINDIR}/BUP_generator.py
-    install -m 0755 ${S}/bootloader/rollback/rollback_parser.py ${D}${BINDIR}
-    sed -i -e's,^#!/usr/bin/python,#!/usr/bin/env python,' ${D}${BINDIR}/rollback_parser.py
+    install -d ${D}${BINDIR}/rollback
+    install -m 0755 ${S}/bootloader/rollback/rollback_parser.py ${D}${BINDIR}/rollback
+    sed -i -e's,^#!/usr/bin/python,#!/usr/bin/env python,' ${D}${BINDIR}/rollback/rollback_parser.py
```

Comment:
In master, the script just goes into `${BINDIR}` instead of a subdirectory.  Any reason for changing this?

Reply:
The reason I move it into `${BINDIR}/rollback` folder, is because I want to run `doflash.sh` and `tegra186-flash-helper.sh` in the generated `core-image-tx2-minimal-jetson-tx2i.tegraflash.zip` to sign BUP-payload.

The l4t_bup_gen.func used by the `tegra186-flash-helper.sh` assume the path `/rollback/rollback_parser.py`. here is the function:

l4t_bup_gen.func line 30-39:
```
_parse_rollback_info()
{
	local tegra_soc="${1}";
	local _rollback_bin=rollback.bin
	local _rollback_cfg="${OUT}"/rollback/"${tegra_soc}"/rollback.cfg
	if [ -f "${_rollback_bin}" ]; then
		rm "${_rollback_bin}"
	fi
	"${OUT}"/rollback/rollback_parser.py "${PRODUCT_NAME}" "${_rollback_cfg}"
}
```

Reply:
OK, I see.  There's still no need to move rollback_parser.py here - if you look at the line you are also changing in image_types_tegra.bbclass, you'll see that the symlink that gets created goes into the expected `rollback` subdirectory.

To make it possible to do BUP generation from a tegraflash package, there are other changes needed, though.

Reply:
Hi @madisongh I revert the this changes back, bitbake can generate payload becasue of oe_make_bup_payload_common() creates symbolic link (through `ln -sf ${STAGING_BINDIR_NATIVE}/tegra186-flash/rollback_parser.py ./rollback/`)

But the final tegraflash.zip does not have `rollback_parser.py` in rollback folder. Is it OK to put `rollback_parser.py` in the rollback folder instead? So that we can run BUP-payload off bitbake?